rq/group.py

```diff
@@ -0,0 +1,121 @@
+from typing import List, Optional, Union
+from uuid import uuid4
+
+from redis import Redis
+from redis.client import Pipeline
+
+from . import Queue
+from .exceptions import NoSuchGroupError
+from .job import Job
+from .queue import EnqueueData
+from .utils import as_text
+
+
+class Group:
+    """A Group is a container for tracking multiple jobs with a single identifier."""
+
+    REDIS_BATCH_NAME_PREFIX = 'rq:group:'
+    REDIS_BATCH_KEY = 'rq:groups'
+
+    def __init__(self, connection: Redis, id: str = None):
+        self.id = id if id else str(uuid4())
+        self.connection = connection
+        self.key = '{0}{1}'.format(self.REDIS_BATCH_NAME_PREFIX, self.id)
+
+    def __repr__(self):
+        return "Group(id={})".format(self.id)
+
+    def _add_jobs(self, jobs: List[Job], pipeline: Optional['Pipeline'] = None):
```

Comment:
Since this is an internal method, let's make the `pipeline` argument required.

Reply:
Done